internal/plugin/index_builder.go

```diff
@@ -173,3 +193,53 @@ func (i *IndexBuilder) appendIndexEntry(entries map[string][]pluginBinariesIndex
 
 	return nil
 }
+
+const jsonSchemaSpecURL = "https://json-schema.org/draft-07/schema"
+
+func (i *IndexBuilder) validateJSONSchemas(in Index) error {
+	schemaLoader := gojsonschema.NewReferenceLoader(jsonSchemaSpecURL)
+	schemaDraft07, err := gojsonschema.NewSchema(schemaLoader)
+	if err != nil {
+		return fmt.Errorf("while loading JSON schema draft-07: %w", err)
+	}
+
+	errs := multierror.New()
+	for _, entry := range in.Entries {
+		entrySchemaLoader := i.getJSONSchemaLoaderForEntry(entry)
+		if err != nil {
+			wrappedErr := fmt.Errorf("while loading JSON schema for %s: %w", entry.Name, err)
+			errs = multierror.Append(errs, wrappedErr)
+			continue
+		}
+		if entrySchemaLoader == nil {
+			continue
+		}
+
+		jsonSchemaValidationResult, err := schemaDraft07.Validate(entrySchemaLoader)
+		if err != nil {
+			wrappedErr := fmt.Errorf("while validating JSON schema for %s: %w", entry.Name, err)
+			errs = multierror.Append(errs, wrappedErr)
+			continue
+		}
+
+		for _, err := range jsonSchemaValidationResult.Errors() {
+			wrappedErr := fmt.Errorf("while validating JSON schema for %s: %s", entry.Name, err)
```

Comment:
The sentence might be confusing, since here the validation operation is already succeeded, but there are validation errors. What about saying `fmt.Errorf("%s schema is invalid. %w", entry.Name, err)` ?

Reply:
👍 Good idea, I changed that to:
```
wrappedErr := fmt.Errorf("invalid schema %q: %s", entry.Name, err)
```
+ added additional test case. Hope that's fine for you 👍

Reply:
Thanks for the prompt reaction ❤️